sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py

```diff
@@ -30,25 +30,60 @@
 from apache_beam.metrics.metricbase import MetricName
 
 
-# TODO(pabloem)(JIRA-1381) Implement this once metrics are queriable from
-# dataflow service
 class DataflowMetrics(MetricResults):
   """Implementation of MetricResults class for the Dataflow runner."""
 
-  def __init__(self, dataflow_client=None, job_result=None):
+  def __init__(self, dataflow_client=None, job_result=None, job_graph=None):
     """Initialize the Dataflow metrics object.
 
     Args:
       dataflow_client: apiclient.DataflowApplicationClient to interact with the
         dataflow service.
       job_result: DataflowPipelineResult with the state and id information of
-        the job
+        the job.
+      job_graph: apiclient.Job instance to be able to translate between internal
+        step names (e.g. "s2"), and user step names (e.g. "split").
     """
     super(DataflowMetrics, self).__init__()
     self._dataflow_client = dataflow_client
     self.job_result = job_result
     self._queried_after_termination = False
     self._cached_metrics = None
+    self._job_graph = job_graph
+
+  def _translate_step_name(self, internal_name):
+    """Translate between internal step names (e.g. "s1") and user step names."""
+    if not self._job_graph:
+      raise ValueError('Could not translate the internal step name.')
+
+    try:
+      [step] = [step
+                for step in self._job_graph.proto.steps
+                if step.name == internal_name]
+      [user_step_name] = [prop.value.string_value
+                          for prop in step.properties.additionalProperties
+                          if prop.key == 'user_name']
+    except ValueError:
+      raise ValueError('Could not translate the internal step name.')
```

Comment:
Is there value in exposing the underlying exception? For example it could fail because 0 values or 2 or more values to unpack and each will indicate a different problem.